mls-rs/src/group/mod.rs

```diff
@@ -1827,13 +1826,19 @@ where
         &mut self.state
     }
 
-    fn can_continue_processing(&self, provisional_state: &ProvisionalState) -> bool {
-        !(provisional_state
-            .applied_proposals
-            .removals
-            .iter()
-            .any(|p| p.proposal.to_remove == self.private_tree.self_index)
-            && self.pending_commit.is_none())
+    fn removal_proposal(
+        &self,
+        provisional_state: &ProvisionalState,
+    ) -> Option<ProposalInfo<RemoveProposal>> {
+        match self.pending_commit {
```

Comment:
It's not clear to me why this test is done.

Reply:
It wasn't clear to me either, but it is there to ensure that external commits work. Maybe @mulmarta has a better idea of why this works, because I forget

Reply:
I think for external commit, `private_tree.self_index = 0` at this point because it has to be something, but you don't know what it should be before you apply the proposals. So if the external committer removes its old copy, and the old copy has index `0`, this would make the external committer fail.

I will have to double check though.

Reply:
Thank you for the answers. How are these indexes related to checking if there's a pending commit?

Reply:
Ok @mulmarta I think if there is some issue there or better way to detect this situation we can patch it after as what is there now is equal to what was there before. Let us know if you find anything out.